mm2src/mm2_main/src/lp_swap.rs

```diff
@@ -1182,15 +1217,50 @@ async fn broadcast_my_swap_status(ctx: &MmArc, uuid: Uuid) -> Result<(), String>
     };
     status.hide_secrets();
 
+    let status = SwapStatus {
+        method: "swapstatus".into(),
+        data: SwapStatusData::Legacy(Box::new(status)),
+    };
+    let msg = json::to_vec(&status).expect("Swap status ser should never fail");
+
     #[cfg(not(target_arch = "wasm32"))]
-    try_s!(save_stats_swap(ctx, &status).await);
+    {
+        // Extract status back. We are avoiding cloning it since it's a big object.
+        let SwapStatusData::Legacy(status) = status.data else {
+            unreachable!("We are sure that status is Legacy variant");
+        };
+
+        try_s!(save_stats_swap(ctx, *status).await);
+    }
+
+    broadcast_p2p_msg(ctx, swap_topic(&uuid), msg, None);
+    Ok(())
+}
 
+async fn broadcast_my_v2swap_status(ctx: &MmArc, status: TPUSwapStatusForStats) -> Result<(), String> {
+    let uuid = status.uuid;
+
+    // Serialize the status to prepare for broadcasting
     let status = SwapStatus {
         method: "swapstatus".into(),
-        data: status,
+        data: SwapStatusData::Tpu(Box::new(status)),
     };
     let msg = json::to_vec(&status).expect("Swap status ser should never fail");
+
+    #[cfg(not(target_arch = "wasm32"))]
+    {
+        // Extract status back. We are avoiding cloning it since it's a big object.
```

Comment:
Worth for warning people (with doc-comment) to not to add "Clone" on the data-type.

Reply:
[ca87640](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2645/commits/ca87640b5df2a1725c83b2d21a284baebef3c124)